models/buffer.go

```diff
@@ -32,19 +48,29 @@ type Buffer struct {
 }
 
 // NewBuffer returns a new empty Buffer with the given capacity.
-func NewBuffer(name string, alias string, capacity int) *Buffer {
+func NewBuffer(name string, alias string, capacity int, strategy string, path string) Buffer {
+	bm := NewBufferMetrics(name, alias, capacity)
+
+	if strategy == StrategyDisk {
+		return NewDiskBuffer(name, capacity, path, bm)
+	} else if strategy == StrategyOverflow {
+		// todo implementme
+		// todo log currently unimplemented
+		return NewMemoryBuffer(capacity, bm)
+	} else if strategy == StrategyMemory || strategy == "" {
+		return NewMemoryBuffer(capacity, bm)
+	}
```

Comment:
I would use a switch statement with the strings instead of defining variables for the strings, so you can directly see the option-value/code correspondence here
```suggestion
	switch strategy {
	case "", "memory": {
		return NewMemoryBuffer(capacity, bm)
	case "disk":
		return NewDiskBuffer(name, capacity, path, bm)
	case "overflow":
		// todo implementme
		// todo log currently unimplemented
		return NewMemoryBuffer(capacity, bm)
	}
```